context/AuthProvider.js

```diff
@@ -0,0 +1,19 @@
+import { createContext, useState } from "react";
+import useLocalStorage from "../../Hooks/useLocalStorage";
+
+const AuthContext = createContext({});
+
+export const AuthProvider =({children}) =>{
+    const [auth, setAuth] = useState({});
+    const [userName, setUserName] = useLocalStorage("userName", '');
+    const [chooseClass, setChooseClass] = useLocalStorage("chooseClass", '');
+    const [traineeEmail, setTraineeEmail] = useLocalStorage("traineeEmail", '');
+
+    return(
+        <AuthContext.Provider value={{auth, setAuth, userName, setUserName, chooseClass, setChooseClass, traineeEmail, setTraineeEmail}}>
```

Comment:
try to move this

{auth, setAuth, userName, setUserName, chooseClass, setChooseClass, traineeEmail, setTraineeEmail}

into an object and set the object as context value. easier to read that way.